File: src/utils/data_structures/finding_shortest_path/shortest_path.py
def Dijkstras(g: Graph), start_index: int -> list:
    cost: list = [math.inf] * g.num_nodes
    last: list [-1] * g.num_nodes
    pq: PriorityQueue = PriorityQueue(min_heap=True)
    
    pq.enqueue(start_index, 0.0)
    for i in range(g.num_nodes):
        if i != start_index:
            pq.enqueue(i, math.inf)
    cost[start_index] = 0.0
    
    while not pq.is_empty():
        index: int = pq.dequeuq()
        
        for edge in g.nodes[index].get_edge_list():
            neighbor: int = edge.to_node
            
            if pq.in_queue(neighbor):
                new_cost: float = cost[index] + edge.weight
                
                if new_cost < cost[neighbor]:
                    pq.update_priority(neighbor, new_cost)
                    last[neighbor] = index
                    cost[neighbor] = new_cost
                    
    return last

File: src/utils/data_structures/finding_shortest_path/shortest_path.ts
function Dijkstras(g: any, start_index: any) {
  const cost = new Array(g.num_nodes).fill(Infinity);
  const last = new Array(g.num_nodes).fill(-1);
  const pq = new PriorityQueue(true);
  
  pq.enqueue(start_index, 0.0);
  for (let i = 0; i < g.num_nodes; i++) {
    if (i !== start_index) {
      pq.enqueue(i, Infinity);
    }
  }
  cost[start_index] = 0.0;
  
  while (!pq.isEmpty()) {
    const index = pq.dequeue();
    
    for (const edge of g.nodes[index].getEdgeList()) {
      const neighbor = edge.to_node;
      
      if (pq.inQueue(neighbor)) {
        const new_cost = cost[index] + edge.weight;
        
        if (new_cost < cost[neighbor]) {
          pq.updatePriority(neighbor, new_cost);
          last[neighbor] = index;
          cost[neighbor] = new_cost;
        }
      }
    }
  }

  return last;
}